lib/_app/AppBar/AppBar.tsx

```diff
@@ -1,63 +1,108 @@
-import {
-  AppBar as MuiAppBar,
-  Container,
-  IconButton,
-  Typography,
-} from '@mui/material';
-import React, { useState } from 'react';
 import Image from 'next/image';
-import { Icon } from '~/lib/components/Icon';
-import SearchIcon from '~/public/images/icons/search-regular.svg';
-import TimesIcon from '~/public/images/icons/times-regular.svg';
-import s from './AppBar.module.scss';
 import Link from 'next/link';
+import React, { useState } from 'react';
+import { useRouter } from 'next/router';
+import clsx from 'clsx';
 
 export const AppBar = () => {
   const [showSearch, setShowSearch] = useState(false);
+  const router = useRouter();
+  const isHomepage = router.pathname == '/';
```

Comment:
You were using `router.pathname == '/'` repeatedly in several places. I thought it made sense to create a re-usable variable for it.